ext/gd/gd.c

```diff
@@ -1055,6 +1055,34 @@ PHP_FUNCTION(imagecopyresampled)
 }
 /* }}} */
 
+#if defined(HAVE_GD_COPY_ROTATED)
+PHP_FUNCTION(imagecopyrotated)
+{
+	zval *SIM, *DIM;
+	zend_long SX, SY, SW, SH, A;
+	gdImagePtr im_dst, im_src;
+	int srcH, srcW, srcY, srcX, angle;
+	double dstX, dstY;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "OOddlllll", &DIM, gd_image_ce, &SIM, gd_image_ce, &dstX, &dstY, &SX, &SY, &SW, &SH, &A) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	im_dst = php_gd_libgdimageptr_from_zval_p(DIM);
+	im_src = php_gd_libgdimageptr_from_zval_p(SIM);
+
+	srcX = SX;
+	srcY = SY;
+	srcH = SH;
+	srcW = SW;
+	angle = A;
+
+	gdImageCopyRotated(im_dst, im_src, dstX, dstY, srcX, srcY, srcW, srcH, angle);
+	RETURN_TRUE;
+}
```

Comment:
It doesn't really make any sense to take the destination Image as an argument, as I frankly imagine one will not have pre allocated such an object.

How do other functions from the GD extension work?

```suggestion
PHP_FUNCTION(imagecopyrotated)
{
	zval *SIM;
	zend_long SX, SY, SW, SH, A;
	gdImagePtr im_dst, im_src;
	int srcH, srcW, srcY, srcX, angle;
	double dstX, dstY;

	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Oddlllll", &SIM, gd_image_ce, &dstX, &dstY, &SX, &SY, &SW, &SH, &A) == FAILURE) {
		RETURN_THROWS();
	}

	im_dst = create_new_object_here();
	im_src = php_gd_libgdimageptr_from_zval_p(SIM);

	srcX = SX;
	srcY = SY;
	srcH = SH;
	srcW = SW;
	angle = A;

	gdImageCopyRotated(im_dst, im_src, dstX, dstY, srcX, srcY, srcW, srcH, angle);
	RETURN_NEW_OBJECT();
}
```

EDIT: ~~Okay I see that other GD functions already do this, but am I the only one to find this stupid or?~~ In any case if the function can only fail when throwing an exception make the function void.